src/components/contentAvail/contentAvailSrv.js

```diff
@@ -1,160 +1,230 @@
+                        return resultsFreeContent;
+                    });
                 }
 
-                return _baseIsEntityAvail().then(function(res) {
-                    if (res === true) {
-                        return true;
-                    }
+                 function isSectionAvail(examId,sectionId){
+                    return _baseIsEntityAvail().then(function(res){
+                        if(res === true){
+                            return true;
+                        }
+
+                        var purchaseData = res[0];
+                        var freeContent = res[1];
+                        var specials = res[2];
 
-                    var tutorialKeyInStorage = idToKeyInStorage(tutorialId);
+                        var examKeyProp = idToKeyInStorage(examId);
+                        var sectionKeyProp = idToKeyInStorage(sectionId);
 
-                    var purchaseData = res[0];
-                    var freeContent = res[1];
+                        var isExamPurchased = _isExamPurchased(purchaseData,examId);
+                        if(isExamPurchased ){
+                            return true;
+                        }
 
-                    if(freeContent.tutorial[tutorialKeyInStorage]){
-                        return true;
+                        var resultsFreeContent = _isFreeContent(freeContent,['exam',examKeyProp,'sections',sectionKeyProp]);
+
+                        if(specials.exam) {
+                            return (resultsFreeContent || _isFreeContent(specials,['exam',examKeyProp,'sections',sectionKeyProp]));
+                        }
+
+                        return resultsFreeContent;
+                    });
+                }
+
+                 function isTutorialAvail(tutorialId){
+                    if(isNaN(tutorialId)){
+                        return $q.reject('ContentAvailSrv: tutorial id should be a number');
                     }
 
-                    return !!(purchaseData.tutorial === PURCHASED_ALL || purchaseData.tutorial[tutorialKeyInStorage]);
+                    return _baseIsEntityAvail().then(function(res) {
+                        if (res === true) {
+                            return true;
+                        }
 
-                });
+                        var tutorialKeyInStorage = idToKeyInStorage(tutorialId);
 
+                        var purchaseData = res[0];
+                        var freeContent = res[1];
+                        var specials = res[2];
 
-            };
+                        if(freeContent.tutorial[tutorialKeyInStorage] || specials.tutorial[tutorialKeyInStorage]){
+                            return true;
+                        }
+
+                        return !!(purchaseData.tutorial === PURCHASED_ALL || purchaseData.tutorial[tutorialKeyInStorage]);
+
+                    });
+                }
+
+                // api
+                return {
```

Comment:
Dont implement it this way because this way the context (this) is not reachable inside the sevice